openedx/core/djangoapps/user_api/views.py

```diff
@@ -156,33 +184,38 @@ def post(self, request):
         if data.get("honor_code") and "terms_of_service" not in data:
             data["terms_of_service"] = data["honor_code"]
 
+    def _get_third_party_auth_redirect_url(self, request):
```

Comment:
**Note to reviewers:** I added a redirect_url to the new registration response in order to make old tests pass.  I am currently testing removal of this (for now) to see if this addition is actually needed in this cleanup task.

Reply:
Testing removal of `redirect_url` in this PR: https://github.com/edx/edx-platform/pull/22090

Reply:
This is the only weird part of this PR, because everything is about removal (or refactor), but this introduces a change to an existing API that is already in use.  

It would be cleanest if this code weren't needed in this PR.  If it is, I would comment the code (in addition to the comments in the PR) to clarify that this code is not used and is just required temporarily for old tests that will be removed.

Reply:
Update: The [third_party_auth...test_full_pipeline_succeeds_registering_new_account](https://github.com/edx/edx-platform/blob/b3421c85802881be05f67b7debe0155dfc022475/common/djangoapps/third_party_auth/tests/specs/base.py#L784) test currently requires the `redirect_url` field in the registration response.

Reply:
I’m missing something. We had two views and you are removing The deprecated one that had this field, but the newer did not.

Does that mean we had a bug in the newer and just never tested it? Or is this testing something that we don’t need? Or other?